src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarDeserializationSchema.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.pulsar.internal;
+
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+
+import org.apache.pulsar.client.api.Message;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+/**
+ * The deserialization schema describes how to turn the Pulsar reader messages
+ * into data types (Java/Scala objects) that are processed by Flink.
+ *
+ * @param <T> The type created by the keyed deserialization schema.
+ */
+public interface PulsarDeserializationSchema<T> extends Serializable, ResultTypeQueryable<T> {
+
+    /**
+     * Method to decide whether the element signals the end of the stream. If
+     * true is returned the element will not be emitted.
+     *
+     * @param nextElement The element to check for the end-of-stream signal.
+     * @return True, if the element signals end of stream, false otherwise.
+     */
+    boolean isEndOfStream(T nextElement);
```

Comment:
Do you have an example for this interface? It would be easier for me to understand why do we need this interface?

Reply:
I added an example to the description above for your reference.

Reply:
On the other hand, the consideration is that Flink’s `DeserializationSchema`(default interface) implements `isEndOfStream` and `deserialize` to regulate the processing behavior of the data source. From a compatibility perspective, I think we should also follow this specification.

Reply:
> Do you have an example for this interface? It would be easier for me to understand why do we need this interface?

There is a similar implementation in `flink-connector-kafka-base`, [KafkaDeserializationSchema.java](https://github.com/apache/flink/blob/5be27b090c80e97c611db83e04b5d82a8abced9d/flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaDeserializationSchema.java)

Add the `PulsarDeserializationSchema` interface, which allows users to obtain the Properties information in the Message.
I also needed this scenario before, and needed to get the data of Message#getProperties(). Can be solved through this interface